scripts/validation.js

```diff
@@ -0,0 +1,58 @@
+function isValidEmail(email) {
+  const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
+  return emailRegex.test(email);
+}
+
+function isValidPasswordLength(password, minLength = 8) {
+  return password.length >= minLength;
+}
+
+function isNotEmpty(value) {
+  return value.trim() !== "";
+}
+
+function isMatching(value1, value2) {
+  return value1 === value2;
+}
+
+function showError(inputElement, message) {
+  const container = inputElement.closest(".input-item");
+
+  const existingError = container.querySelector('[class*="text-error"]');
+  if (existingError) {
+    existingError.remove();
+  }
+
+  inputElement.classList.add("input-invalid");
+
+  const errorSpan = document.createElement("span");
+  errorSpan.className = "text-error text-sm block mt-2 font-semibold";
+  errorSpan.textContent = message;
+
+  if (inputElement.parentElement.classList.contains("input-wrapper")) {
+    inputElement.parentElement.parentElement.appendChild(errorSpan);
+  } else {
+    container.appendChild(errorSpan);
+  }
+}
```

Comment:
💊 제안
이미 에러 메시지 태그가 존재할때 지우고 다시 생성하시는 것은 DOM 조작을 여러번하게됩니다. 
에러가 존재한다면 내부 문자열만 바꿔주는 방식이 성능 및 가독성에도 더 좋을 것 같습니다.

```suggestion
function showError(inputElement, message) {
  const container = inputElement.closest(".input-item");

  let errorSpan = container.querySelector(".text-error");
  
  // error 태그 생성하는 부분은 함수로 분리해도 좋을듯 합니다.
  if (!errorSpan) {
    errorSpan = document.createElement("span");
    errorSpan.className = "text-error text-sm block mt-2 font-semibold";
    container.appendChild(errorSpan);
  }

  inputElement.classList.add("input-invalid");
  errorSpan.textContent = message;
}
```